libbeat/docs/outputconfig.asciidoc

```diff
@@ -196,10 +196,27 @@ For more information, see <<configuring-ingest-node>>.
 
 ===== pipelines
 
-Array of pipeline selector configurations supporting conditionals, format string
-based field access and name mappings. The first rule matching will be used to
-set the `pipeline` for the event to be published. If `pipelines` is missing or no
-rule matches, the `pipeline` field will be used.
+Similar to the `indices` array, this is an array of pipeline selector
+configurations supporting conditionals, format string based field access
+and name mappings. The first rule matching will be used to set the
+`pipeline` for the event to be published. If `pipelines` is missing or
+no rule matches, the `pipeline` field will be used.
+
+Example elasticsearch output with `pipelines`:
+
+["source","yaml"]
+------------------------------------------------------------------------------
+output.elasticsearch:
+  hosts: ["http://localhost:9200"]
+  index: "logs-%{+yyyy.MM.dd}"
+  pipelines:
+    - pipeline: critical_pipeline
+      when.contains:
+        type: "critical"
```

Comment:
I am actually not completely sure where this field can be declared. I assume it comes from the fields declaration in the prospector section so a complete example (which I agree we should add) should look something like:

```
filebeat.prospectors:
- paths: ["/var/log/app/normal/*.log"]
  fields:
    type: "normal"
- paths: ["/var/log/app/critical/*.log"]
  fields:
    type: "critical"

output.elasticsearch:
  hosts: ["http://localhost:9200"]
  index: "filebeat-%{+yyyy.MM.dd}"
  pipelines:
    - pipeline: critical_pipeline
      when.contains:
        type: "critical"
    - pipeline: normal_pipeline
      when.contains:
        type: "normal"
```

Instead of ```when.contains`` using ```when.equals``` would probably be better here. Does that look okay?